### ISSUE TITLE
distributed_table data distribution too slow

### ISSUE THREAD

**Issue by wuhongda** [ClickHouse/ClickHouse#50125]
clickhouse :)  SELECT version()
┌─version()─┐
  │ 21.8.14.5 │
└───────────┘

clickhouse :)   SELECT *FROM system.distribution_queue

┌─database─┬─table────────┬─data_path───────────────────────────────────────────────────────────────────────────┬─is_blocked─┬─error_count─┬─data_files─┬─data_compressed_bytes─┬─broken_data_files─┬─broken_data_compressed_bytes─┬─last_exception─┐
│ jaeger   │ jaeger_index │ /rds/clickhouse/store/ce2/ce27e528-cf6a-47e1-91bc-2d7b61f9ed1d/shard7_all_replicas/ │          0 │           0 │          1 │                123825 │                 0 │                            0 │                │

clickhouse :)  SELECT database,name,engine, total_bytesFROM system.tables
WHERE isNotNull(total_bytes) AND (database NOT IN ('system')) AND (engine = 'Distributed') AND (total_bytes != 0)
ORDER BY total_bytes ASC
┌─database─┬─name─────────┬─engine──────┬──total_bytes─┐
│ wuhongda   │ wuhongda_spans │ Distributed │       262046 │
│ wuhongda   │ wuhongda_index │ Distributed │ 216504206894 │
└──────────┴──────────────┴─────────────┴──────────────┘

clickhouse :)  SHOW SETTINGS LIKE '%background%'
┌─name─────────────────────────────────────────┬─type───┬─value─┐
│ background_buffer_flush_schedule_pool_size   │ UInt64 │ 16    │
│ background_pool_size                         │ UInt64 │ 256   │
│ background_move_pool_size                    │ UInt64 │ 8     │
│ background_fetches_pool_size                 │ UInt64 │ 8     │
│ background_schedule_pool_size                │ UInt64 │ 256   │
│ background_message_broker_schedule_pool_size │ UInt64 │ 16    │
│ background_distributed_schedule_pool_size    │ UInt64 │ 256   │
└──────────────────────────────────────────────┴────────┴───────┘

Question:

Data in distributed tables is piled up. After parameter adjustment, the load of underlying nodes does not increase significantly, and no error log is reported.

**Comment by den-crane** [comment]
21.8.14.5  is out of support . Upgrade to 23.4.

Distributed monitor is single-threaded in some cases it's unable to keep-up with ingestion.

Try to enable sync mode - https://clickhouse.com/docs/en/operations/settings/settings#insert_distributed_sync

**Comment by wuhongda** [comment]
> 

thanks！

**Comment by wuhongda** [comment]
> 21.8.14.5 is out of support . Upgrade to 23.4.
> 
> Distributed monitor is single-threaded in some cases it's unable to keep-up with ingestion.
> 
> Try to enable sync mode - https://clickhouse.com/docs/en/operations/settings/settings#insert_distributed_sync
thanks！This data has already been written with async.
And now I need to get rid of the accumulated data before I can upgrade.

**Comment by wuhongda** [comment]
So now i need to see what went wrong and fix it.

**Comment by den-crane** [comment]
>thanks！This data has already been written with async.

I see only 1 file `data_files: 1`

```
┌─database─┬─table────────┬─data_path───────────────────────────────────────────────────────────────────────────┬─is_blocked─┬─error_count─┬─data_files─┬─data_compressed_bytes─┬─broken_data_files─┬─broken_data_compressed_bytes─┬─last_exception─┐
│ jaeger │ jaeger_index │ /rds/clickhouse/store/ce2/ce27e528-cf6a-47e1-91bc-2d7b61f9ed1d/shard7_all_replicas/ │ 0 │ 1592381 │ 1 │ 216486426837 │ 0 │ 0 │ │
```

**Comment by wuhongda** [comment]
Yes, we have 6 shards, 2 replicate per shard, now all shard stacked distributed table data is shard num 4，and send slow

**Comment by wuhongda** [comment]
[root@clickhouse wuhongda_index]# pwd
/rds/clickhouse/data/wuhongda/wuhongda_index
[root@clickhouse wuhongda_index]# du -sch *
51M	shard1_all_replicas
51M	shard3_all_replicas
97G	shard4_all_replicas
51M	shard5_all_replicas
51M	shard6_all_replicas
51M	shard7_all_replicas
97G	总用量

ps: The wuhongda_index is distributed table,and the shard4_all_replicas don't send or send slow
